plat/socionext/synquacer/drivers/scmi/sq_scmi.c

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2019, ARM Limited and Contributors. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <assert.h>
+
+#include <arch_helpers.h>
+#include <common/debug.h>
+
+#include <drivers/arm/css/scmi.h>
+#include <drivers/arm/css/css_mhu_doorbell.h>
+#include <plat/arm/common/plat_arm.h>
+#include <plat/common/platform.h>
+
+#include <sq_common.h>
+
+#include "sq_scmi_private.h"
+#include "sq_scmi.h"
+
+const uint32_t plat_css_core_pos_to_scmi_dmn_id_map[PLATFORM_CORE_COUNT] = {
+	0,   1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11,
+	12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23
+};
+
+#define SQ_SCMI_PAYLOAD_BASE		PLAT_SQ_SCP_COM_SHARED_MEM_BASE
+#define MHU_CPU_INTR_S_SET_OFFSET	0x308
+
+static scmi_channel_plat_info_t sq_scmi_plat_info = {
+		.scmi_mbx_mem = SQ_SCMI_PAYLOAD_BASE,
+		.db_reg_addr = PLAT_CSS_MHU_BASE + MHU_CPU_INTR_S_SET_OFFSET,
+		.db_preserve_mask = 0xfffffffe,
+		.db_modify_mask = 0x1,
+		.ring_doorbell = &mhu_ring_doorbell,
+};
+
+scmi_channel_plat_info_t *plat_css_get_scmi_info()
+{
+	return &sq_scmi_plat_info;
+}
+
+void plat_arm_gic_cpuif_disable(void)
+{
+	sq_gic_cpuif_disable();
+}
+
+void plat_sq_pwrc_setup(void)
+{
+	plat_arm_pwrc_setup();
+}
+
+struct dram_info_resp {
+	int status;
+	int reserved;
+	struct draminfo info;
+};
```

Comment:
This structure can go in scmi.h at the end where a section for "vendor-specific" extensions can be created.

Reply:
I have submitted updated commit.
I really hesitate to add this definitions into include/drivers/arm/css/scmi.h.
 # I also need to add very vendor specific "struct draminfo" definition here.

Instead of modifying scmi.h, I added drivers/arm/css/scmi/vendor/scmi_sq.h.
This file contains synquacer specific protocol definition.

If you prefer to modify scmi.h, I will do so.

Reply:
ok, what you have done is fine.